CMakeLists.txt

```diff
@@ -9,20 +9,16 @@ if(PROJECT_BINARY_DIR STREQUAL PROJECT_SOURCE_DIR)
    message(FATAL_ERROR "In-tree build attempt detected, aborting. Set your build dir outside your source dir, delete CMakeCache.txt from source root and try again.")
 endif()
 
-option(basic_warnings "Basic compiler warnings." ON)
 option(private_dbus "Use private dbus namespace to prevent clashes during development." OFF)
 
 include(cmake/coverage.cmake)
 include(FindPkgConfig)
 include (GNUInstallDirs)
 
-set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} --std=c++11")
 add_definitions(-DQT_NO_KEYWORDS)
 
-if(${basic_warnings})
-  set(CMAKE_C_FLAGS "${CMAKE_C_FLAGS} -Wall -pedantic")
-  set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -Wall -pedantic")
-endif()
+set(CMAKE_C_FLAGS "${CMAKE_C_FLAGS} -Wall -pedantic -Werror -Wextra")
+set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} --std=c++11 -Wall -pedantic -Werror -Wextra")
```

Comment:
We should `set(CMAKE_CXX_STANDARD 11)` for the C++ standard, instead of adding it to `CMAKE_CXX_FLAGS`.